template-manager-cmtemplate/src/main/resources/definitions/cm-cloud-storage-location-specification.json

```diff
@@ -4,7 +4,7 @@
       "propertyName": "zeppelin.notebook.s3.bucket",
       "propertyFile": "zeppelin-site",
       "description": "The directory where Zeppelin notebooks are saved",
-      "defaultPath": "{{{ defaultPath }}}/zeppelin/notebook",
```

Comment:
I would like to keep it to "{{{ defaultPath }}}/zeppelin/notebook, because if you remove it, then by default the notbooks go into the root directory, which feels a bit ugly

Reply:
Actually, as per [the related Zeppelin S3NotebookRepo class](https://github.com/apache/zeppelin/blob/master/zeppelin-plugins/notebookrepo/s3/src/main/java/org/apache/zeppelin/notebook/repo/S3NotebookRepo.java) and [AWS Java SDK](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/model/PutObjectRequest.html), this config should really contain only the bucket name, no key path at all. Maybe it would be best to enforce this via some validation. All path segments (i.e. object key prefix) should instead go to property `zeppelin.notebook.s3.user`.

Reply:
@akanto yes zeppelin needs `zeppelin.notebook.s3.bucket `to be initialized only by bucket name. Hence `zeppelin.notebook.s3.user` is initialized  to <clustername>/zeppelin. so final notebook path becomes something like <bucket>/<cluster>/zeppelin/notebook